server/plugin/api_test.go

```diff
@@ -63,6 +63,157 @@ func TestWithRecovery(t *testing.T) {
 	}
 }
 
+func TestPlugin_handleSkip(t *testing.T) {
+	httpTestJSON := testutils.HTTPTest{
+		T:       t,
+		Encoder: testutils.EncodeJSON,
+	}
+
+	for name, test := range map[string]struct {
+		httpTest          testutils.HTTPTest
+		request           testutils.Request
+		expectedResponse  testutils.ExpectedResponse
+		userID            string
+		parseAuthTokenErr error
+		loadUserErr       error
+		callError         error
+	}{
+		"Skip message is successfully sent to virtual Agent": {
+			httpTest: httpTestJSON,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("%s%s", pathPrefix, constants.PathSkip),
+				Body: model.PostActionIntegrationRequest{
+					ChannelId: "mock-channelID",
+				},
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+			userID: "mock-channelID",
+		},
+		"User is unauthorized": {
+			httpTest: httpTestJSON,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("%s%s", pathPrefix, constants.PathSkip),
+				Body: model.PostActionIntegrationRequest{
+					ChannelId: "mock-channelID",
+				},
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusUnauthorized,
+			},
+			userID: "",
+		},
+		"Error while decoding response body": {
+			httpTest: httpTestJSON,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("%s%s", pathPrefix, constants.PathSkip),
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+			userID: "mock-channelID",
+		},
+		"User is not present in store": {
+			httpTest: httpTestJSON,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("%s%s", pathPrefix, constants.PathSkip),
+				Body: model.PostActionIntegrationRequest{
+					ChannelId: "mock-channelID",
+				},
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+			userID:      "mock-userID",
+			loadUserErr: errors.New("error in loading the user from KVstore"),
+		},
+		"Error occurs while parsing OAuth token": {
+			httpTest: httpTestJSON,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("%s%s", pathPrefix, constants.PathSkip),
+				Body: model.PostActionIntegrationRequest{
+					ChannelId: "mock-channelID",
+				},
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+			userID:            "mock-userID",
+			parseAuthTokenErr: errors.New("error while parsing OAuth token"),
```

Comment:
Instead of all this, do something like this - https://github.com/Brightscout/mattermost-plugin-servicenow/blob/e9d9ae34f20bc52872e7ab022924b582208b4ea1/server/plugin/api_test.go#L45

and this - https://github.com/Brightscout/mattermost-plugin-servicenow/blob/e9d9ae34f20bc52872e7ab022924b582208b4ea1/server/plugin/api_test.go#L1699